src/broker/shard_parser.c

```diff
@@ -318,7 +317,15 @@ sp_make_int_sp_value_from_string (SP_VALUE * value_p, char *pos, int length)
   int result = 0;
   char tmp = pos[length];
 
-  pos[length] = '\0';
+  if (length > 1 && pos[length - 1] == ';')
```

Comment:
I think you need to check for cases where semicolons occur multiple times.
example)
/*+ shard_key */ 1;;

Reply:
You right, multiple semicolons in shard key should be cleared.
Thank you.